service/src/test/java/bio/terra/workspace/service/workspace/WorkspaceServiceTest.java

```diff
@@ -247,23 +264,43 @@ void testUpdateWorkspace() {
     String otherDescription = "The deprecated workspace";
 
     Workspace secondUpdatedWorkspace =
-        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, null, otherDescription, null);
+        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, null, null, otherDescription, null);
 
     // Since name is null, leave it alone. Description should be updated.
+    assertEquals(userFacingId, secondUpdatedWorkspace.getUserFacingId().orElse(null));
     assertEquals(name, secondUpdatedWorkspace.getDisplayName().orElse(null));
     assertEquals(otherDescription, secondUpdatedWorkspace.getDescription().orElse(null));
     assertEquals(propertyMap2, updatedWorkspace.getProperties());
 
     // Sending through empty strings and an empty map clears the values.
     Map<String, String> propertyMap3 = new HashMap<>();
     Workspace thirdUpdatedWorkspace =
-        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, "", "", propertyMap3);
+        workspaceService.updateWorkspace(USER_REQUEST, workspaceId, "", "", "", propertyMap3);
```

Comment:
It would be more robust to test with separate strings, so we don't have a false negative if we cross wires somewhere.

Reply:
Hmm, I made this test more realistic by passing `userFacingId` instead of `""`. (Controller validates that `userFacingId` can't be `""`. This test doesn't call controller, which is why `""` works.) So there's no more assertion for userFacingId.